src/test/java/com/fittracker/fittracker/controller/WeightControllerIntegrationTest.java

```diff
@@ -95,25 +98,78 @@ void givenValidPostRequestWithExistingDate_shouldReturnError() throws Exception
-            weightRepository.save(new Weight(LocalDate.of(2023, 8, 3), 13.2, testUUID));
-            assertThat(weightRepository.findAll()).hasSize(1);
+        @Nested
+        class WeightEndpoint {
+            @Test
+            void givenValidGetRequestWithExistingDate_shouldReturnWeight() throws Exception {
+                weightRepository.save(new Weight(LocalDate.of(2023, 8, 3), 13.2, testUUID));
 
-            mockMvc.perform(get(URI.create(ENDPOINT + "?date=2023-08-03"))
-                            .header("Authorization", "Bearer " + TOKEN))
-                    .andExpect(status().isOk())
-                    .andExpect(content().string("{\"date\":\"2023-08-03\",\"value\":13.2}"));
+                mockMvc.perform(get(URI.create(ENDPOINT + "?date=2023-08-03"))
+                                .header("Authorization", "Bearer " + TOKEN))
+                        .andExpect(status().isOk())
+                        .andExpect(content().string("{\"date\":\"2023-08-03\",\"value\":13.2}"));
+
+                assertThat(weightRepository.findAll()).hasSize(1);
+            }
+
+            @Test
+            void givenValidGetRequestWithNonexistentDate_shouldReturnError() throws Exception {
+
+                mockMvc.perform(get(URI.create(ENDPOINT + "?date=2023-08-03"))
+                                .header("Authorization", "Bearer " + TOKEN))
+                        .andExpect(status().isNotFound())
+                        .andExpect(content().string("{\"message\":\"Weight not found for date: 2023-08-03\"}"));
+            }
         }
 
-        @Test
-        void givenValidGetRequestWithNonexistentDate_shouldReturnError() throws Exception {
+        @Nested
+        class WeightsEndpoint {
+            private final ObjectMapper mapper = new ObjectMapper();
+
+            @BeforeEach
+            public void beforeEach() {
+                List<Weight> weights = List.of(
+                        new Weight(LocalDate.of(2023, 1, 1), 100.0, testUUID),
+                        new Weight(LocalDate.of(2023, 1, 10), 95.5, testUUID),
+                        new Weight(LocalDate.of(2023, 2, 20), 98.2, testUUID)
+                );
+                weights.forEach(weight -> weightRepository.save(weight));
+            }
+
+            @Test
+            void givenDateRange_shouldReturnListOfWeightResponses() throws Exception {
+                mockMvc.perform(get(URI.create(ENDPOINT + "s?startDate=2023-01-01&endDate=2023-03-04"))
+                                .header("Authorization", "Bearer " + TOKEN))
+                        .andExpect(status().isOk())
+                        .andExpect(content().string("[{\"date\":\"2023-01-01\",\"value\":100.0}," +
+                                "{\"date\":\"2023-01-10\",\"value\":95.5}," +
+                                "{\"date\":\"2023-02-20\",\"value\":98.2}]"));
+
+                assertThat(weightRepository.findAll()).hasSize(3);
```

Comment:
I think we don't need these assertions

Reply:
We are assuring here that GET request is idempotent (at least that id doesn't change amount of elements in a database). I would Leave it as we have these assertions in other GET methods.

Reply:
To not repet it in every test I put this in the @AfterEach.

Reply:
It would make sense for endpoints in which we alter the database, but I'd get rid of it for GET endpoints, as we're only reading the database (which we have covered on unit test level)